pkg/types/alibabacloud/platform.go

```diff
@@ -0,0 +1,13 @@
+package alibabacloud
+
+// Platform stores all the global configuration that all machinesets use.
+type Platform struct {
+	// Region specifies the Alibaba Cloud region where the cluster will be created.
+	Region            string `json:"region"`
+	ResourceGroupName string `json:"resourceGroupName"`
```

Comment:
As `ResourceGroupName` is optional it should be `omitempty`

Reply:
The `ResourceGroupName` is not optional.